_articles/scripting/Typescript/typescript-events.md

```diff
@@ -0,0 +1,187 @@
+---
+title: Events and Timers in Typescript
+author: Shush
+steamId: 76561197994333648
+date: 05.10.2022
+---
+
+As you may know, Dota has many events. While developing a custom game, listening to events is very useful, as it allows you to do something when something occurs. For example, an event may listen to all deaths, triggering whenever a hero, unit or building are killed. Events will supply some information about the instance of that event. For example, in the above event, the killer and the victim will be included in the parameters.
```

Comment:
"an event may listen to all deaths" sounds wrong, the custom game can listen to all death events, the event itself does not do any listening